library/std/src/panic.rs

```diff
@@ -296,9 +296,12 @@ pub fn get_backtrace_style() -> Option<BacktraceStyle> {
     }
 
     // Setting environment variables for Fuchsia components isn't a standard
-    // or easily supported workflow. For now, display backtraces by default.
+    // or easily supported workflow. For now, display backtraces by default,
+    // and allow turning off completely for compiler tests.
     let format = if cfg!(target_os = "fuchsia") {
-        BacktraceStyle::Full
+        crate::env::var_os("RUST_BACKTRACE")
+            .map(|x| if &x == "0" { BacktraceStyle::Off } else { BacktraceStyle::Full })
+            .unwrap_or(BacktraceStyle::Full)
```

Comment:
What happens if we enable `BacktraceStyle::Short`, does it give us the same output? I'm wondering if we can use the code path below and only use target_os to determine the default (`.unwrap_or(BacktraceStyle::Off)` below).

Reply:
I just tested, and BacktraceStyle::Short **does** give a different, shorter backtrace than BacktraceStyle::Full. To say, I think the Short backtrace is functional.

For keeping behavior as similar as possible, do we just want to default to Full, and otherwise allow Short via `RUST_BACKTRACE=1`?

Reply:
Works for me!